dispatcher_plugin_nb2workflow/queries.py

```diff
@@ -10,23 +10,13 @@
 from .dataserver_dispatcher import NB2WDataDispatcher
 from cdci_data_analysis.analysis.ontology import Ontology
 import os
-from functools import lru_cache, wraps
-from json import dumps
+from functools import lru_cache
 from copy import deepcopy
-
-class HashableDict(dict):
-    def __hash__(self):
-        return hash(dumps(self, sort_keys=True))
-
-def with_hashable_dict(func):
-    @wraps(func)
-    def wrapper(backend_param_dict, ontology_path):
-        return func(HashableDict(backend_param_dict), ontology_path)
-    return wrapper
+from .util import with_hashable_dict
 
 @with_hashable_dict
 @lru_cache
-def construct_parameter_lists(backend_param_dict, ontology_path):
+def construct_parameter_lists(bk_descript_dict = {}, ontology_path = None):
```

Comment:
Need to think about it. I added this change to express the fact that these are kwargs. This is because I made `@with_hashable_dict` more generic.

We already discussed a bit that the case of empty/non-accessible/etc ontology probably needs to be treated in the Ontology implementation in oda_api. That's probably the right time to actually do it.

Reply:
We have an issue opened for this. I mark this resolved.